### PR TITLE
June 2022 update: new datasets, jet/MET level validation, additional loss terms

### PR DESCRIPTION
**Pull request by jpata**
- additional stats (50k -> 100k) for each dataset, uploaded to CERN
  - switch QCD from 13->14 TeV as for the other datasets
  - additional QCD_HighPt dataset
- additional CMSSW validation scripts: `mlpf/tallinn/validate_cms.sh`
- additional 2D histogram loss component
- cluster and match jets in the evaluation stage
- save/load the optimizer state to enable a smooth continuation of the training

### CODE REVIEW

File: mlpf/tfmodel/model_setup.py [hunk 1]
```diff
@@ -65,9 +65,9 @@ class ModelOptimizerCheckpoint(tf.keras.callbacks.ModelCheckpoint):
     def on_epoch_end(self, epoch, logs=None):
         super(ModelOptimizerCheckpoint, self).on_epoch_end(epoch, logs=logs)
         weightfile_path = self.opt_path.format(epoch=epoch+1, **logs)
-        try:
+        try:
```
Comment (erwulff) [inline mlpf/tfmodel/model_setup.py:68]: Have we implemented the loading of these optimizer weights somewhere?

**Review by erwulff — COMMENTED** [review]
A couple of very minor comments.

File: mlpf/tfmodel/model_setup.py (hunk 1 above)
Comment (jpata) [reply mlpf/tfmodel/model_setup.py:68]: we use it for restarting/continuing the training, and it's confirmed that it works.

File: parameters/delphes.yaml [hunk 2]
```diff
@@ -223,7 +223,8 @@ train_test_datasets:
     datasets:
       - delphes_pf
 
-validation_dataset: delphes_pf
+validation_dataset:
```
[the remark this answers is not in the document]
Comment (jpata) [reply parameters/delphes.yaml]: yep, already caught it

File: mlpf/tfmodel/model_setup.py (hunk 1 above)
Comment (jpata) [reply mlpf/tfmodel/model_setup.py:68]: https://github.com/jpata/particleflow/blob/june_2022/mlpf/pipeline.py#L248
